### ISSUE TITLE
APK broken (signature)

### ISSUE THREAD

**Issue by IzzySoft** [mRahulJain/TodoApp_Android#1]
Just to notify you that something must have gone wrong with your signing process:

```
DOES NOT VERIFY
ERROR: SHA-256 digest of classes.dex does not match the digest specified in META-INF/MANIFEST.MF. Expected: <UONG2Xkjg4j7rEZs294JVhVkiVAl2y8W2LVgjYh4DGw=>, actual: <rHlq1JckI3QKdzsLNwp3muOyQSbf5DNbu771388vVMA=>
ERROR: SHA-256 digest of res/drawable-v21/design_password_eye.xml does not match the digest specified in META-INF/MANIFEST.MF. Expected: <OLRbf1aLfQbIV2jDvqZPWXL4evY4pOHpWHq8CvdpWZI=>, actual: <KNQEhPzA9gBZQKYakY/q089XycHfmLCWN1PGSaTwfAM=>
...
```

You can check that using `apksigner verify app-release.apk`.

**Comment by mRahulJain** [comment]
I don't get it. The given apk file installed perfectly on two of my smart phones. So, what actually is going wrong ?

**Comment by IzzySoft** [comment]
Good question. Let me try again manually, maybe the download was broken (but then, it would be an invalid ZIP file, right?)

```
$ apksigner verify app-release.apk |grep -v WARN
DOES NOT VERIFY
ERROR: SHA-256 digest of classes.dex does not match the digest specified in META-INF/MANIFEST.MF. Expected: <UONG2Xkjg4j7rEZs294JVhVkiVAl2y8W2LVgjYh4DGw=>, actual: <rHlq1JckI3QKdzsLNwp3muOyQSbf5DNbu771388vVMA=>
ERROR: SHA-256 digest of res/drawable-v21/design_password_eye.xml does not match the digest specified in META-INF/MANIFEST.MF. Expected: <OLRbf1aLfQbIV2jDvqZPWXL4evY4pOHpWHq8CvdpWZI=>, actual: <KNQEhPzA9gBZQKYakY/q089XycHfmLCWN1PGSaTwfAM=>
ERROR: SHA-256 digest of res/layout-v21/notification_template_custom_big.xml does not match the digest specified in META-INF/MANIFEST.MF. Expected: <2TP55V6vxuBojdGM+tetNuA7AG04VUFYkGQthp+8KfE=>, actual: <9HyCAy5zmjDW33ZUFwa6hp76ACaFzPrQ9RgCWrdoDgk=>
ERROR: SHA-256 digest of res/layout-v21/notification_template_icon_group.xml does not match the digest specified in META-INF/MANIFEST.MF. Expected: <RDCDzASDJKxHc73NcRxYkay0CmNrgBFLcss5Tggv+3Q=>, actual: <8vbp3P08yN5H/o3DAVPJzFMqL8EcXp5qT4duS3py1ZE=>
...
```

I didn't try whether it installs (IMHO it should not because signature validation should fail on-device as well) – but usually this picture says "be careful, looks like someone has tampered with the APK" (like, unzip, modify, repack-without-resign).

If you could (build and) sign again, I can try with the new APK if you let me have it. Thanks!

PS: Instead of putting the APK inside the repo, you might consider [tagging releases](https://github.com/mRahulJain/Todo-Notes_App_Android/releases) and attaching it there. Doesn't "bloat" the repo that way (so potential contributors have less to clone) while still providing the APK to the public.

**Comment by mRahulJain** [comment]
It was just a simple SQLite based project of mine. So, keeping the apk open won't be a big issue. Still, I will be removing it soon. Thanks..

Also the apk provided now works well. 👍

**Comment by IzzySoft** [comment]
Oh, please do not completely REmove but rather move (as described above) – or I can no longer provide [your app via my repo](https://apt.izzysoft.de/fdroid/index/apk/com.example.todo). Maybe the next version won't have that error anymore, and fdroidserver doesn't reject it (as it does with the current one).

**Comment by mRahulJain** [comment]
Heyyy!! You can't use simply someone's app right ?? Can I know, with whose permission did you upload it in your repo ?

**Comment by IzzySoft** [comment]
I just removed it¹. Missed that you didn't specify a license – and you're right, open source doesn't mean *free* and open source. So I respect you don't want it distributed and made recoverable.

¹ <sub>change will become effective with the next sync in about 4h</sub>
